### PR TITLE
Temp perm inf

### PR DESCRIPTION
**Pull request by ev0x**
- bans where a tempban is in place will clear the tempban
- on ban or tempban infractions such as mutes etc will be set to inactive
- on guild join unban and kick infractions will be cleared (You can only join a guild if your not banned)
- report command author dodgy unicode handled and set to snowflake

### CODE REVIEW

**Comment by ev0x** [comment]
this one needs `frestart` as it includes a bugfix in core.py

File: rowboat/plugins/infractions.py [hunk 1]
```diff
@@ -1351,25 +1316,102 @@ def softban(self, event, user, reason=None):
...
+                    reason = reason.split(' ', 1)[-1]
+                else:
+                    reason = None
+        elif duration:
+            duration = parse_duration(duration)
+
+        member = None
+        act = False
+
+        u = Object()
```
Comment (TobiTenno) [inline rowboat/plugins/infractions.py]: target_user instead of `u`?

File: rowboat/plugins/infractions.py [hunk 2]
```diff
@@ -1351,25 +1316,102 @@ def softban(self, event, user, reason=None):
...
+        if member is None:
+            u.id = user
+            member = Object()
+            member = int(to_snowflake(u))
+            act = True
+
         if member:
-            self.can_act_on(event, member.id)
-            expires_dt = parse_duration(duration)
+            g = None
```
Comment (TobiTenno) [inline rowboat/plugins/infractions.py]: guild_ban instead of `g`?

**Comment by ev0x** [comment]
updated as mentioned plus forgot to add the updated user model

**Pull request merged**